src/intro/no-std.md

```diff
@@ -1,5 +1,74 @@
 # A `no_std` Rust Environment
 
-> ❌: **This section should cover:**
->
-> * what #[no_std] means: no I/O (filesystems), allocation, etc.
\ No newline at end of file
+As mentioned earlier, this book is all about "bare metal" programming. Standard libraries in turn usually
+require some sort of system (os) integration and higher level functionality, on top of which they implement their functionality.
+Such functionalities are for example I/O (files, networking), Threads, and Heap (dynamic memory allocation).
```

Comment:
Not quite accurate. Embedded systems are really all about I/O, just a different kind and level of I/O than typical developers are thinking of.

Reply:
Yeah I'd change that to `are for example file I/O, network sockets, threads and dynamic memory allocation`.